src/components/CreateVmWizard/steps/BasicSettings.js

```diff
@@ -226,17 +230,46 @@ class BasicSettings extends React.Component {
   // Calculate the proper number of virtual sockets, cores per virtual socket, threads per core according to the number of virtual CPUs
   // any change of the number of vCpus in the Basic settings requires recalculation of the numer of cores, sockets, threads
   getTopologySettings (vcpus, force = {}) {
+    const { maxNumOfSockets, maxNumOfCores, maxNumOfThreads } = this.grabCpuOptions()
     return getTopology({
       value: vcpus,
       max: {
-        sockets: this.props.maxNumOfSockets,
-        cores: this.props.maxNumOfCores,
-        threads: this.props.maxNumOfThreads,
+        sockets: maxNumOfSockets,
+        cores: maxNumOfCores,
+        threads: maxNumOfThreads,
       },
       force: force,
     })
   }
 
+  grabCpuOptions () {
+    const { data, clusters, templates } = this.props
+
+    // No cluster has been selected yet.........
+    if (!isValidUid(data.clusterId)) {
+      return {
+        maxNumOfSockets: 1,
+        maxNumOfCores: 1,
+        maxNumOfThreads: 1,
+        maxNumOfVmCpus: 1,
+      }
+    }
+
+    // CPU options come from the selected Template (if defined there) or Cluster
+    const cluster = clusters.get(data.clusterId)
+    const clusterOptions = cluster.get('cpuOptions')
+
+    const template = templates.get(data.templateId)
+    const templateOptions = template ? template.get('cpuOptions') : new Map()
+
+    return {
+      maxNumOfSockets: templateOptions.get('maxNumOfSockets') || clusterOptions.get('maxNumOfSockets'),
```

Comment:
>  1. are we OK with having a mix of cluster and template? i.e. if template is missing one value

Not really, but it can't happen anyway.

> 
>   2. do we need a fallback to default values? i.e. as proposed above by providing defaults via `getIn` ?

The Cluster will always have the values defined, so that IS the fallback.

> 
>   3. do we need those props to be undefined ? if not then we could merge using spread operator and pick the props i.e.
>        `const {maxNumofCores} = {...clusterOptions, ...templateOptions}`

No.  I'll adjust things a bit to make it easier to read.